### ISSUE TITLE
a11y tests don't run on the production URL

### ISSUE THREAD

**Issue by shawnbot** [18F/college-choice#1502]
As you can see in [this build](https://circleci.com/gh/18F/college-choice/324), it looks like the accessibility tests are failing right off the bat on the master branch. They fail on being able to load the production URL:

```
npm run test-a11y

> college-choice-js@0.0.0 test-a11y /home/ubuntu/college-choice
> ./test/a11y.js

Error: Page "https://collegescorecard.ed.gov/" could not be opened
  (stack trace)
```

You can simulate these conditions locally with:

```sh
CIRCLE_BRANCH=master npm run test-a11y
```

On my machine this runs fine. Maybe a PhantomJS issue? Unclear.

**Comment by LisaGee** [comment]
@Shawnbot - But in the previous release, they all passed?

**Comment by shawnbot** [comment]
@LisaGee that's correct. In the previous release, though, the CI tests were running their own copy of the site. In #1474 I made the suggestion that we should run the browser (and accessibility) tests off Federalist because it's faster and less brittle. So far that's been true, and I didn't have any reason to suspect that it would break like this because I tested the production URL with this process locally.

I can tap @ultrasaurus or @yozlet to pair with me on figuring this out, but I don't see it as a show-stopper. It appears to be a bug somewhere deep in the libraries that we're using or an issue with the SSL certificate that the site uses, because the a11y tests fail immediately and the stack trace comes from a library called [truffler](https://github.com/springernature/truffler), not from the tools that we're using directly. (I did some informal testing and confirmed that truffler doesn't like our production URL, but that's about where my knowledge ends.)

**Comment by shawnbot** [comment]
So I dug a bit deeper into this and discovered that PhantomJS (that's the headless browser that our a11y test suite uses to load the pages and run the accessibility testing tools) does have [some problems with SSL](https://github.com/ariya/phantomjs/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+%22SSL+handshake+failed%22), and I think this could be the issue.

This source of the problem is pretty far upstream (our tests -> pa11y -> truffler -> PhantomJS), and I wasn't able to sidestep PhantomJS's low-level problems using any of the suggested [workarounds](https://github.com/ariya/phantomjs/issues/12440) while SSH'd into a Circle CI build. For future reference, my test was a PhantomJS script that just loads the URL and prints the status:

```js
var page = require('webpage').create();

page.open('https://collegescorecard.ed.gov', function(status) {
  console.log('status:', status);
  phantom.exit();
});
```

Then you run this with:

```sh
phantomjs test.js
```

I might need an SSL wizard like @konklone to help me figure out what the problem is here, but the solution on our end either need to be a change to our test configuration, an upgrade to the underlying libraries, or even a new SSL certificate. :rage4:

**Comment by LisaGee** [comment]
@Shawnbot - Is option 4 just to put it back to the way it was before the change, since it seemed to have worked? Or is the speed really an impediment?

**Comment by konklone** [comment]
If it's a TLS issue ("SSL" is so passé now), I'm gonna guess it's Server Name Indication, which some earlier versions of phantomjs choke on.

It looks from the build like it's using Phantomjs 1.9.19. Though I can't find the exact commit or documentation, I looked into this over the summer and I believe Phantomjs 2.0 [added better SNI support](https://github.com/macbre/phantomas/issues/488#issuecomment-117820655). Can you update the stack to use a 2.x version of Phantomjs?

**Comment by shawnbot** [comment]
That's a great call, @konklone. I'll see if I can get Circle CI to use a 2.x build of PhantomJS.

**Comment by shawnbot** [comment]
And yes, @LisaGee, reverting is always an option. :)

**Comment by shawnbot** [comment]
This wasn't supposed to auto-close, but we have a [passing build](https://circleci.com/gh/18F/college-choice/336) on master now, so it's all good!

**Comment by shawnbot** [comment]
Thanks @konklone for the suggestion to upgrade PhantomJS. That was what did it.

**Comment by konklone** [comment]
Woo!!
